Userland/Libraries/LibJS/Runtime/TypedArrayPrototype.cpp

```diff
@@ -23,6 +23,10 @@ void TypedArrayPrototype::initialize(GlobalObject& object)
 
     // FIXME: This should be an accessor property
     define_native_property(vm.names.length, length_getter, nullptr, Attribute::Configurable);
+    define_native_property(vm.names.buffer, buffer_getter, nullptr, Attribute::Configurable);
+    define_native_property(vm.names.byteLength, byte_length_getter, nullptr, Attribute::Configurable);
+    define_native_property(vm.names.byteOffset, byte_offset_getter, nullptr, Attribute::Configurable);
+    define_native_property(vm.names.BYTES_PER_ELEMENT, bytes_per_element_getter, nullptr, 0); // FIXME: This should just be a normal property and not a getter.
```

Comment:
This should not be on `TypedArrayPrototype`, but on each individual TypedArray prototype. yes, that's confusing :D

Each TypedArray has its own prototype object (e.g. `Uint8ArrayPrototype`), which is https://tc39.es/ecma262/#sec-properties-of-typedarray-prototype-objects in the spec and generated in the `JS_DECLARE_TYPED_ARRAY` / `JS_DEFINE_TYPED_ARRAY` macros. They all share some functionality via an internal `TypedArrayPrototype` - this thing here. In the spec that's `%TypedArray.prototype%`.

With all that being said, implementing this correctly is easy as we get the C++ value type in that macro. In fact we already do this on each constructor - just copy this to the prototype.

https://github.com/SerenityOS/serenity/blob/ebdeed087c2438f6a58e4e9e4495ef6e4bbe8d5f/Userland/Libraries/LibJS/Runtime/TypedArray.cpp#L211